bin/chandra_repro

```diff
@@ -2532,8 +2546,65 @@ def remake_fov1_file( pars ):
     v5("skyfov returned")
     v5(out)
 
+def process_sso(pars, is_gratings=False):
+    """
+    Run sso_freeze to add solar system object coordinates to the 
+    event file and the aspect solution file.    
+    """
+    from ciao_contrib.runtool import sso_freeze
 
+    sso_eph = pars["sso_file"]
+    if sso_eph is None:
+        return
 
+    if is_gratings:
+        evt="evt1a_file"
+        out_evt1 = os.path.join(pars["outdir"], pars["root"] + '_repro_sso_evt1a.fits')
+    else:
+        evt="evt1_file"
+        out_evt1 = os.path.join(pars["outdir"], pars["root"] + '_repro_sso_evt1.fits')
+
+    evt1=pars[evt]
+    eph1=pars["eph1_file"]
+    asol1="@"+pars["asol1_file"]
+        
+    if len(eph1) == 0:
+        v0("WARNING: Could not find definitive orbit ephemeris file, orbit*eph1.fits, skipping sso_freeze")
+
+    tab = pyc.read_file(sso_eph)
+    obj = tab.get_key_value("OBJECT")        
+    v1("\nRunning sso_freeze to add solar system coordinates for object: "+obj)
+    
+    out_asol1 = os.path.join(pars["outdir"], pars["root"] + '_repro_sso_asol1.fits')
+
+    sso_freeze.infile = evt1
+    sso_freeze.scephemfile = eph1
+    sso_freeze.ssoephemfile = sso_eph
+    sso_freeze.asolfile = asol1
+    sso_freeze.outfile = out_evt1
+    sso_freeze.ocsolfile = out_asol1
+    sso_freeze.clobber = pars["clobber"]
+    sso_freeze.verbose = pars["verbose"]
+    out = sso_freeze()
+    
+    v5("sso_freeze returned")
+    v5(out)
+    
+    # It would be nice to set the ASOLFILE keyword to the sso asol
+    # since it has the same format. Unfortunately, the sso asol 
+    # has blockname="ocsol" instead of "ASPSOL" and that causes
+    # down stream tools (asphist) to error out.  Leave this 
+    # here for the day that's fixed.
+    # ~ tab = pyc.read_file(out_evt1, mode="rw")
+    # ~ tab.get_key("ASOLFILE").value = os.path.basename(out_asol1)
+    # ~ tab.write()
```

Comment:
Could we just change the block name of the file (or write out a copy of it with the name ASPSOL)? I think crates should allow is to do this, but maybe there's other things to worry about?

Reply:
So there are two things 

I tried to change the `.name`, but doing that causes crates to write out a new extension to the file -- not update the existing block.  So instead of updating the existing file I'd have to write out a temporary file and deal with that.  Which seems like more trouble than it's worth.

The 2nd reason is that w/ this failing in so early w/ `asphist` it's unclear if there might be other cases further down stream where normal/typical things could go astray because of the extra columns. (There shouldn't be but haven't tested through all the tools that use asol, incl. marx/chart).

Reply:
Awesome...